packages/amplify-provider-awscloudformation/src/iterative-deployment/deployment-manager.ts

```diff
@@ -220,25 +251,29 @@ export class DeploymentManager {
       const service = interpret(machine)
         .onTransition(async state => {
           if (state.changed) {
-            this.spinner.render();
-            if (state.matches('idle')) {
-              this.logSpinnerMessage('rollback', `Starting rollback`);
-            } else if (state.matches('preRollback')) {
-              this.logSpinnerMessage('rollback', `Waiting for previous deployment to finish`);
-            } else if (state.matches('rollback.waitForTablesToBeReady')) {
-              this.logSpinnerMessage('rollback', `Waiting for DynamoDB indices to be ready`);
-            } else if (state.matches('rollback')) {
-              this.logSpinnerMessage('rollback', `Rolling back (${maxDeployed - state.context.currentIndex} of ${maxDeployed})`);
+            const key = Object.keys(rollbackSpinnerMessage).find(elm => state.matches(elm));
+            if (key) {
+              const keyObj = rollbackSpinnerMessage[key];
+              this.logger(keyObj.machine, [{ spinner: keyObj.message }])();
+              this.spinner.start(keyObj.message);
             } else if (state.matches('rolledBack')) {
-              this.spinner.succeed(`Rolled back succesfully`);
-              this.logger('rollback', [{ spinner: 'Rolled back successfully' }])();
+              this.spinner.stop();
+              this.printer.finishBars();
+              this.printer.stopBars();
+              promptsPrinter.info(`Rolled back (${maxDeployed - state.context.currentIndex} of ${maxDeployed})`);
+              this.logger('rollback', [{ spinner: 'Rolled back successfully' }]);
+            } else if (state.matches('rollback')) {
+              this.spinner.stop();
```

Comment:
should stop implicitly stop bars ? we seem to do a lot of low level operations for something that should look like.
```
spinner = new Spiner()
try {
spinner.start()
} finally {
spinner stop():
}
```

Reply:
So, should spinner be intelligent and detect if it has bars or not?
Would we ever have bars without spinner?
I.e. would this be better design?
```
Bars bars = new Bars();
Spinner spinnerWithBars = new Spinner(bars);
try {
 spinnerWithBars.start()
 } finally {
 spinnerWithBars.stop()
 }
 // and then
 Spinner spinnerWithoutBars = new Spinner()
  try {
 spinnerWithoutBars.start()
 } finally {
 spinnerWithoutBars.stop()
 }
```

If this makes sense but is too big feel free to make it backlog item.